.travis.yml

```diff
@@ -8,26 +8,32 @@ php:
   - 7.0
   - hhvm
 
-matrix:
-  allow_failures:
-    - php: 7.0
-    - php: hhvm
+sudo: false
 
 cache:
   directories:
     - $HOME/.composer/cache
+    - vendor
```

Comment:
> Didn't find the information on how this actually works in Travis doc.

This is not about Travis but about composer itself.

Just run a `composer update` on your project, change a deps version or include a new one on your `composer.json` and run `composer update` again. You will see that only the concerned package will be changed.
This is the same behavior on Travis.

> Your changes on bin dir actually makes the Makefile bugging.

Well, this is related to bin dir configuration so, not cache.

As I said, I'll take a look for that. ;-)